File: 554. Brick Wall/Python/Solution.py
class Solution(object):
    def leastBricks(self, wall):
        """
        :type wall: List[List[int]]
        :rtype: int
        """
        count = {}
        for row in wall:
            current = 0
            for item in row[:-1]:
                current += item
                if current not in count:
                    count[current] = 0
                count[current] += 1
        return len(wall) - (max(count.values()) if count else 0)

File: 554. Brick Wall/C++/Solution.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>> &wall) {
        unordered_map<int, int> count;
        for (auto &row : wall) {
            int current = 0;
            for (int i = 0; i < row.size() - 1; i++) {
                current += row[i];
                if (count.find(current) == count.end()) {
                    count[current] = 0;
                }
                count[current]++;
            }
        }
        int maxValue = 0;
        for (auto &value : count) {
            maxValue = max(maxValue, value.second);
        }
        return wall.size() - maxValue;
    }
};